velox/functions/prestosql/window/AggregateWindow.cpp

```diff
@@ -158,7 +158,7 @@ class AggregateWindowFunction : public exec::WindowFunction {
       lastRow = std::max(lastRow, rawFrameEnds[i]);
       nonDecreasingFrameEnd &= rawFrameEnds[i] >= rawFrameEnds[i - 1];
     }
-    bool fixedFirstRow = firstRow == rawFrameStarts[0];
+    bool fixedFirstRow = rawFrameStarts[0] == rawFrameStarts[numRows - 1];
```

Comment:
Why did you change this ? firstRow is a value checked across all rows so if it is == rawFrameStarts[0] then that means the value is same across all the rows. The new condition only checks if the first and last row have the same frameStarts values. There isn't a guarantee of the values in between.

Reply:
This was changed because when the frame start is currentRow, fixedFirstRow was getting set to true and the fixedStartAggregation function was called, resulting in wrong results. Thanks for pointing out this drawback, it should be fine with the current frames supported but with cases like ```c2 following``` bounds, it could present a problem. Modified fixedFirstRow computation it a bit to include all rows in between. Please let me know if the current check is fine. Thanks!